### PR TITLE
chore: upgrade vitest to ^1.0.4

### PR DESCRIPTION
**Pull request by yonadaaa**

### CODE REVIEW

File: e2e/packages/client-vanilla/package.json [hunk 1]
```diff
@@ -34,6 +34,6 @@
   "devDependencies": {
     "rimraf": "^3.0.2",
     "vite": "^4.2.1",
-    "vitest": "0.31.4"
+    "vitest": "^1.0.4"
   }
-}
+}
```
Comment (holic) [inline e2e/packages/client-vanilla/package.json:39]: hmm we should figure out why this trailing empty line differs as it prob points to an inconsistent linter or formatter

did you edit these files manually or use something like `pnpm upgrade`?

Comment (yonadaaa) [reply e2e/packages/client-vanilla/package.json:39]: manually 😬 couldn't figure out how to specify the version with `pnpm ugprade`

Comment (yonadaaa) [reply e2e/packages/client-vanilla/package.json]: I enabled **Insert Final Newline** in my VSCode and the formatting is correct now

File: packages/world/ts/config/worldConfig.test-d.ts
Comment (yonadaaa) [inline packages/world/ts/config/worldConfig.test-d.ts:1]: Maybe there's some way we can still use [`toEqualTypeOf`](https://vitest.dev/api/expect-typeof.html#toequaltypeof`) here. Failing test run: https://github.com/latticexyz/mud/actions/runs/7654925617/job/20859876029?pr=2190#step:6:1856

**Comment by yonadaaa** [comment]
Closing in favour of https://github.com/latticexyz/mud/pull/2193

**Pull request closed without merging**